tests/std/tests/P0896R4_ranges_alg_remove/test.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <span>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+using namespace std;
+using P = pair<int, int>;
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::remove(borrowed<false>{}, 42)), ranges::dangling>);
+STATIC_ASSERT(same_as<decltype(ranges::remove(borrowed<true>{}, 42)), ranges::subrange<int*>>);
+
+struct instantiator {
+    static constexpr P expected[3] = {{0, 99}, {2, 99}, {4, 99}};
+
+    template <ranges::forward_range Read>
+    static constexpr void call() {
+        using ranges::remove, ranges::subrange, ranges::iterator_t;
+        { // Validate iterator + sentinel overload
+            P input[5] = {{0, 99}, {1, 47}, {2, 99}, {3, 47}, {4, 99}};
+            Read wrapped_input{input};
+
+            auto result = remove(wrapped_input.begin(), wrapped_input.end(), 47, get_second);
+            STATIC_ASSERT(same_as<decltype(result), subrange<iterator_t<Read>>>);
+            assert(result.begin() == next(wrapped_input.begin(), 3));
+            assert(result.end() == wrapped_input.end());
+            assert(ranges::equal(expected, span<P>{input, 3}));

```

Comment:
I observe that `span{input}.first<3>()` would cover the same elements but have static extent. I don't know that it matters much, but there it is.

Reply:
I might go with `span<P, 3>{input, 3}` for some passive aggressive review response ;)

Reply:
Done